### PR TITLE
Add defensive shield + shield logic

### LINKED ISSUE #259: Crusher
- [x] #317
- [x] #318
- [x] #319

### LINKED ISSUE #319: Crusher abilities
- [x] #364 
- [x] #365 
- [x] #366 
- [x] #367 
- [x] #402 
- [x] #410
- [ ] #411

### LINKED ISSUE #411: Defensive shield

### PR DESCRIPTION
**Pull request by Vermidia**
## About the PR

Adds the crushers defensive shield + logic for other shields when we get them.

## Why / Balance

Implementing a feature, which should make crushers a bit more viable as an evolution.
Resolves #319, resolves #411, resolves #259

## Technical details

Adds the XenoShieldSystem and CrusherShieldSystem, the former handling basic shield logic (eating damage, breaking, adding/removing new shields) and the latter handling crusher specific parts (damping all damage types by 10, handling the action).

Shields are calculated after armor (from what I could tell on cm-13) and can have durations and decay rates.

I also ported the empower effects + the armor bars from cm-13.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/user-attachments/assets/f360bde4-e3a6-419d-8b9e-ee001fa0908c

## Breaking changes

I hope not?

**Changelog**

:cl:
- add: Crushers can now use their defensive shield ability that lasts 7 seconds, which gives them a 200 hitpoint shield that takes 10 less damage from all sources, and for a short while gives them explosion immunity.

### CODE REVIEW

**Comment by KingOnBudget** [comment]
CRUSHER SWEEP
!!!!!

**Comment by Dutch-VanDerLinde** [comment]
Resolves #319

**Comment by Dutch-VanDerLinde** [comment]
also resolves #259

File: Content.Shared/_RMC14/Shields/RemovedShieldEvent.cs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+﻿namespace Content.Shared._RMC14.Shields;
+[ByRefEvent]
```
Comment (DrSmugleaf) [inline Content.Shared/_RMC14/Shields/RemovedShieldEvent.cs:2]: Small style nitpick
```suggestion
﻿namespace Content.Shared._RMC14.Shields;
﻿
[ByRefEvent]
```

File: Content.Shared/_RMC14/Shields/XenoShieldSystem.cs [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            shield.ShieldAmount -= shield.DecayPerSecond * frameTime;
+
+            if (shield.ShieldAmount <= 0)
+            {
+                RemoveShield(uid, shield.Shield);
+                continue;
+            }
+
+			Dirty(uid, shield);
+		}
```
Comment (DrSmugleaf) [inline Content.Shared/_RMC14/Shields/XenoShieldSystem.cs:144]:
```suggestion
            Dirty(uid, shield);
        }
```

File: Content.Shared/_RMC14/Shields/CrusherShieldSystem.cs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+namespace Content.Shared._RMC14.Shields;
+
+public sealed partial class CrusherShieldSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly INetManager _net = default!;
+    [Dependency] private readonly XenoShieldSystem _shield = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly XenoPlasmaSystem _xenoPlasma = default!;
+    public override void Initialize()
```
Comment (DrSmugleaf) [inline Content.Shared/_RMC14/Shields/CrusherShieldSystem.cs:20]:
```suggestion
    [Dependency] private readonly XenoPlasmaSystem _xenoPlasma = default!;

    public override void Initialize()
```

File: Content.Shared/_RMC14/Shields/CrusherShieldSystem.cs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        _popup.PopupEntity(Loc.GetString("rmc-xeno-defensive-shield-activate", ("user", xeno)), xeno, Filter.PvsExcept(xeno), true, PopupType.MediumCaution);
+        _popup.PopupEntity(Loc.GetString("rmc-xeno-defensive-shield-activate-self", ("user", xeno)), xeno, xeno, PopupType.Medium);
+        SpawnAttachedTo(xeno.Comp.Effect, xeno.Owner.ToCoordinates());
+    }
+
+
+    public void ApplyEffects(Entity<CrusherShieldComponent> ent)
+    {
+
+        if (!TryComp<CMArmorComponent>(ent, out var armor))
```
Comment (DrSmugleaf) [inline Content.Shared/_RMC14/Shields/CrusherShieldSystem.cs:56]:
```suggestion
    {
        if (!TryComp<CMArmorComponent>(ent, out var armor))
```

**Review by DrSmugleaf — CHANGES_REQUESTED** [review]
Thanks for the PR!

**Pull request merged**